WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -81,32 +86,33 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
-    /**
-     * intent extras will contain error info if this activity was started from an
-     * upload error notification
-     */
-    private void showErrorDialogIfNeeded(Bundle extras) {
-        if (extras == null || !extras.containsKey(EXTRA_ERROR_MSG) || isFinishing()) {
-            return;
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // don't add search for self-hosted sites
+        if (!mSite.isWPCom()) {
+            return true;
         }
 
-        final String errorMessage = extras.getString(EXTRA_ERROR_MSG);
-
-        if (TextUtils.isEmpty(errorMessage)) {
-            return;
+        getMenuInflater().inflate(R.menu.posts_list, menu);
+
+        mSearchAction = menu.findItem(R.id.search_posts_list);
+        if (mSearchAction != null && mSearchAction.getActionView() != null) {
+            MenuItemCompat.setOnActionExpandListener(mSearchAction, new MenuItemCompat.OnActionExpandListener() {
+                @Override
+                public boolean onMenuItemActionExpand(MenuItem menuItem) {
+                    return true;
+                }
+
+                @Override
+                public boolean onMenuItemActionCollapse(MenuItem menuItem) {
+                    setTitle(getString(mIsPage ? R.string.pages : R.string.posts));
+                    return true;
+                }
+            });
+            ((SearchView) mSearchAction.getActionView()).setOnQueryTextListener(this);
```

Comment:
This is a little bit of a nitpick, but could we move the `(SearchView) mSearchAction.getActionView()` out of the `if` statement, so it doesn't get called twice. (The other one is inside the `if` expression)